react-state/my-app/src/Caption.tsx

```diff
@@ -0,0 +1,13 @@
+import { useState } from 'react';
+
+type Props = {
+  captions: string[];
+};
+export function Caption({ captions }: Props) {
+  const [index, setIndex] = useState(0);
+
+  function handleClick() {
+    setIndex((priIndex) => (priIndex + 1) % captions.length);
```

Comment:
You can use the approach with a function, but it's simpler and more readable to just reference the current state:
`setIndex((index + 1) % captions.length)`

There are a few times where you need the function form of the state setter, but they are rare.